packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,326 @@
+  /// parameter.
+  ///
+  /// The [onChanged] parameter takes a [ValueChanged<String>] which is invoked
+  /// upon a change in the text field's value.
+  ///
+  /// The [onSubmitted] parameter takes a [ValueChanged<String>] which is invoked
+  /// when the keyboard submits.
+  ///
+  /// To provide a hint placeholder text that appears when the text entry is
+  /// empty, pass a [String] to the [placeholder] parameter. This defaults to
+  /// 'Search'.
+  // TODO(DanielEdrisian): Localize the 'Search' placeholder.
+  ///
+  /// The [style] and [placeholderStyle] properties allow changing the style of
+  /// the text and placeholder of the textfield. [placeholderStyle] defaults
+  /// to the gray [CupertinoColors.secondaryLabel] iOS color.
+  ///
+  /// To set the text field's background color and border radius, pass a
+  /// [BoxDecoration] to the [decoration] parameter. This defaults to the
+  /// default translucent tertiarySystemFill iOS color and 9 px corner radius.
+  // TODO(DanielEdrisian): Must make border radius continuous, see
+  // https://github.com/flutter/flutter/issues/13914.
+  ///
+  /// The [itemColor] and [itemSize] properties allow changing the icon color
+  /// and icon size of the search icon (prefix) and X-Mark (suffix).
+  /// They default to [CupertinoColors.secondaryLabel] and [20.0].
+  ///
+  /// The [padding], [prefixInsets], and [suffixInsets] let you set the padding
+  /// insets for text, the search icon (prefix), and the X-Mark icon (suffix).
+  /// They default to values that replicate the `UISearchTextField` look. These
+  /// default fields were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  ///
+  /// To customize the suffix icon, pass an [Icon] to [suffixIcon]. This
+  /// defaults to the X-Mark.
+  ///
+  /// To dictate when the X-Mark (suffix) should be visible, a.k.a. only on when
+  /// editing, not editing, on always, or on never, pass a
+  /// [OverlayVisibilityMode] to [suffixMode]. This defaults to only on when
+  /// editing.
+  ///
+  /// To customize the X-Mark (suffix) action, pass a [VoidCallback] to
+  /// [onSuffixTap]. This defaults to clearing the text.
+  const CupertinoSearchTextField({
+    Key? key,
+    this.controller,
+    this.onChanged,
+    this.onSubmitted,
+    this.style,
+    String this.placeholder = 'Search',
+    this.placeholderStyle,
+    this.decoration,
+    this.backgroundColor,
+    this.borderRadius,
+    this.padding = const EdgeInsets.fromLTRB(3.8, 8, 5, 8),
+    Color this.itemColor = CupertinoColors.secondaryLabel,
+    this.itemSize = 20.0,
+    this.prefixInsets = const EdgeInsets.fromLTRB(6, 0, 0, 4),
+    this.suffixInsets = const EdgeInsets.fromLTRB(0, 0, 5, 2),
+    this.suffixIcon = const Icon(CupertinoIcons.xmark_circle_fill),
+    this.suffixMode = OverlayVisibilityMode.editing,
+    this.onSuffixTap,
+  })  : assert(placeholder != null),
```

Comment:
For all the things that can't be null, also mention that they can't be null in their respective docs.